examples/op/common/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use alloy::providers::ProviderBuilder;
+use anyhow::Context;
+use risc0_zkvm::{sha::Digest, Receipt};
+use url::Url;
+
+pub async fn verify_on_chain(
```

Comment:
The common method gives us an easy way to turn any example into an e2e test:
- It deploys the actual verifier contract on the forked target chain, so we can then do an `eth_call` against it to check that the proof and verification library actually works. 

This has the following drawbacks:
- It adds code to the examples that is not relevant to a user
- Verification requires calls to a public RPC provider, if they are down the tests will fail.

Is this worth it?